ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB/Impl/Snapshots.hs

```diff
@@ -181,18 +182,16 @@ writeExtLedgerState (SomeHasFS hasFS) encLedger path cs = do
 trimSnapshots ::
      Monad m
   => Tracer m (TraceSnapshotEvent r)
-  -> SomeHasFS m
-  -> SomeHasFS m
+  -> [SomeHasFS m]
   -> SnapshotPolicy
   -> m [DiskSnapshot]
-trimSnapshots tracer hasFS ssdhasFS SnapshotPolicy{onDiskNumSnapshots} = do
+trimSnapshots tracer fss SnapshotPolicy{onDiskNumSnapshots} = do
     -- We only trim temporary snapshots
-    diskSnapshots <- filter diskSnapshotIsTemporary <$> listSnapshots hasFS
+    diskSnapshots <- filter diskSnapshotIsTemporary . concat <$> mapM listSnapshots fss
```

Comment:
This is wrong, I need to find a better way to do this. We won't drop enough snapshots as they will be doubled...